android/src/main/java/com/guichaguri/trackplayer/service/MusicService.kt

```diff
@@ -177,7 +178,7 @@ class MusicService : HeadlessJsTaskService() {
 
                 emit(MusicEvents.PLAYBACK_STATE, bundle)
             }
-        }
+            .launchIn(serviceScope)
 
         serviceScope.launch {
             event.audioItemTransition.collect {
```

Comment:
I think we should decide if we should use `collect` or `forEach`, and use one on all observers? @dcvz

Reply:
Ah I actually didn't mean to include this in this PR but yes we should decide, I'd definitely opt for this way though..

but this is purely based on our experience in OneBeat

Reply:
but let's scrap this conversation for now.. let's stick to collect.